Apps/Medication/src/Services/RestMedicationStatementService.cs

```diff
@@ -86,91 +83,47 @@ public async Task<RequestResult<IList<MedicationStatementHistory>>> GetMedicatio
-                return result;
-            }
-        }
-
-        private static (bool Valid, string? ValidationMessage) ValidateProtectiveWord(string? protectiveWord)
-        {
-            bool valid = true;
-            string? errMsg = null;
-            if (!string.IsNullOrEmpty(protectiveWord))
-            {
-                if (protectiveWord.Length is >= MinLengthProtectiveWord and <= MaxLengthProtectiveWord)
+                // Retrieve the phn
+                RequestResult<PatientModel> patientResult = await this.patientService.GetPatient(hdid).ConfigureAwait(true);
+                if (patientResult.ResultStatus != ResultType.Success || patientResult.ResourcePayload == null)
                 {
-                    Regex regex = new(@"^[0-9A-Za-z_]+$");
-                    if (!regex.IsMatch(protectiveWord))
-                    {
-                        valid = false;
-                        errMsg = ErrorMessages.ProtectiveWordInvalidChars;
-                    }
+                    return RequestResultFactory.Error<IList<MedicationStatementHistory>>(patientResult.ResultError);
                 }
-                else
+
+                PatientModel patient = patientResult.ResourcePayload;
+                OdrHistoryQuery historyQuery = new()
+                {
+                    StartDate = patient.Birthdate,
+                    EndDate = DateTime.Now,
+                    Phn = patient.PersonalHealthNumber,
+                    PageSize = 20000,
+                };
+                string ipv4Address = this.httpContextAccessor.HttpContext?.Connection.RemoteIpAddress?.MapToIPv4().ToString() ?? "Unknown";
+                RequestResult<MedicationHistoryResponse> response =
+                    await this.medicationStatementDelegate.GetMedicationStatementsAsync(historyQuery, protectiveWord, hdid, ipv4Address).ConfigureAwait(true);
+
+                if (response.ResultStatus != ResultType.Success || response.ResourcePayload == null)
                 {
-                    valid = false;
-                    errMsg = protectiveWord.Length > MaxLengthProtectiveWord ? ErrorMessages.ProtectiveWordTooLong : ErrorMessages.ProtectiveWordTooShort;
+                    return RequestResultFactory.Error<IList<MedicationStatementHistory>>(patientResult.ResultError);
                 }
-            }
 
-            return (valid, errMsg);
+                IList<MedicationStatementHistory>? payload = this.autoMapper.Map<IList<MedicationStatementHistory>>(response.ResourcePayload.Results);
+                this.PopulateMedicationSummary(payload.Select(r => r.MedicationSummary).ToArray());
+
+                return RequestResultFactory.Success(payload, response.TotalResultCount, response.PageIndex, response.PageSize);
+            }
         }
 
-        private void PopulateMedicationSummary(List<MedicationSummary> medSummaries)
+        private void PopulateMedicationSummary(MedicationSummary[] medSummaries)
```

Comment:
I agree re: IEnumerable - it should be the only list in public methods/dtos, with private it can be any, but preferable IEnumerable as well.
As for array vs list, arrays are immutable and mem contiguous so prob better than Lists IMO.

Reply:
Under the hood, the List class uses an array to store its data, so performance is comparable.

Here's a quote from Eric Lippert's [Arrays Considered Somewhat Harmful](https://learn.microsoft.com/en-us/archive/blogs/ericlippert/arrays-considered-somewhat-harmful) article that expresses why I prefer Lists for internal implementations:

> And finally, given that arrays are mutable by design, the way an array restricts that mutability is deeply weird. All the contents of the collection are mutable, but the size is fixed.  What is up with that? Does that solve a problem anyone actually has?
> 
> For this reason alone I do almost no programming with arrays anymore. Arrays simply do not model any problem that I have at all well – I rarely need a collection which has the rather contradictory properties of being completely mutable, and at the same time, fixed in size. If I want to mutate a collection it is almost always to add something to it or remove something from it, not to change what value an index maps to.

Reply:
Fair enough, however, you're right that the instances in an array are mutable, but in most cases, a code that works on an array does not try to mutate the internals, but the sequence itself, and array do not have .Add() functionality.
In the spirit of functional programming, my personal preference is to end up with linq/side-effect-free functions that operate on IEnumerable.